hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -303,6 +305,26 @@ public Result inputSplits(
     }
   }
 
+  /**
+   * Returns the incremental Hoodie source split batch.
+   *
+   * @param metaClient    The meta client
+   * @param startInstant  The start Instant of the splits
+   * @param cdcEnabled    Whether cdc is enabled
+   *
+   * @return The list of incremental input splits or empty if there are no new instants
+   */
+  public HoodieContinuousSplitBatch inputHoodieSourceSplits(
+      HoodieTableMetaClient metaClient,
+      @Nullable String startInstant,
+      boolean cdcEnabled) {
+    Result result = inputSplits(metaClient, startInstant, cdcEnabled);
+    List<HoodieSourceSplit> splits = result.inputSplits.stream().map(split -> new HoodieSourceSplit(
```

Comment:
wondering why we must have a new `HoodieSourceSplit`, can we rename the old one and add the function you want there?

Reply:
the current supported Flink release all have this interface right? can we impl the `SourceSplit` directly for the old split class?